### PR TITLE
Use a single, dedicated process

### PR DESCRIPTION
**Pull request by dwcoates**
`pgn-mode` now refers to a single Python process, `pgn-mode--python-process`, which runs a single python scripy `pgn_handler.py` that accepts messages on stdin in the form of `*opcode* -- *pgn string*`. E.g., 

```
1 -- [Event "?"]
[Site "?"]
[Date "????.??.??"]
[Round "?"]
[White "?"]
[Black "?"]
[Result "1-0"]
[Missed_mates_found "5"]

1. d4 e6 2. g3 b6 3. bg2 nc6 4. Bf4 Nf6 5. c4 d5 6. cxd5 Nxd4 7. Qxd4 Nxd5 8.
Bxd5 exd5 9. Nc3 Be6 10. Nf3 f6 11. Rd1 Bc5 12. Qd3 c6 13. e4 dxe4 14. Qxd8+
Rxd8 15. Rxd8+ Kxd8 
```

To merge onto `RW/fen-at-point`

### CODE REVIEW

File: pgn-mode.el [hunk 1]
```diff
@@ -271,23 +271,23 @@
...
 
-;; TODO: check for pyhon-chess
 ;; TODO: generalize script
 ;; TODO: pipes?
 (defun pgn-mode--make-process (&optional force)
   "Initialize pgn-mode `pgn-mode--python-process', optionally FORCE recreation if already exists."
+  (pgn-mode-python-chess-guard)
   (when (and (not force) (pgn-mode--process-running-p))
     (error "The pgn-mode Python process already running. Use optional `force' to recreate"))
-  (when (not (= 0 (call-process pgn-mode-python-path nil nil nil "-c" "import chess")))
```
Comment (dwcoates) [inline pgn-mode.el]: Why is this removed?

Comment (rolandwalker) [reply pgn-mode.el]: It's replaced with `(pgn-mode-python-chess-guard)`.
